### DISCUSSION TITLE
posts/ansible-automation/

### DISCUSSION THREAD

**Discussion by giscus** [techno-tim/techno-tim.github.io#211] [Comments]
# posts/ansible-automation/

Documentation for Techno Tim YouTube videos for all your copy pasta needs!

https://technotim.live/posts/ansible-automation/

**Comment by trekie86** [comment]
How do you handle local dns to ensure your inventory can work properly? I'm using PiHole for local DNS but often times my Eero router will allow for local dns queries. With the VMs from ProxMox, those don't resolve using the local lan namespace.

I wrote a script to synchronize my VMs to PiHole but it relies upon the qemu guest agent being installed first which presents a chicken and egg problem. I can't use ansible "easily" to install the QEMU agent and I can't update PiHole with the local DNS entries without QEMU to get the IP of the VM. I could always log into the console and grab the ip address from there but I'm trying to minimize the console needs.

**Comment by jacflesher** [comment]
total newb here, only been working with kubernetes like two weeks, but im on a mission to learn.  how many of these lines are actual action items?  am i supposed to run the commands as part of the install?  where is the "hosts" file you are sharing?

**Comment by R3-AnThRaX** [comment]
you should add the following to the qemu-guest-agent.yml playbook:

    - name: Make sure qemu-guest-agent is started
      systemd:
        name: qemu-guest-agent.service
        state: started
      become: true

That way the qemu-guest-agent is started immediately after installation

**Comment by HeneryH** [comment]
The python versioning and requiring an environment activation now makes this a whole separate layer of learning a new prerequisite before even starting the original task.
